static/app/components/events/eventTagsAndScreenshot/tags.tsx

```diff
@@ -23,13 +25,33 @@ type Props = {
   projectSlug: Project['slug'];
 };
 
+function filterTags(tags: EventTag[], tagFilter: TagFilter): EventTag[] {
+  switch (tagFilter) {
+    case TagFilter.ALL:
+      return tags;
+    case TagFilter.CUSTOM:
+      return tags.filter(tag => !SentryDefaultTags.has(tag.key));
+    default:
+      return tags.filter(tag => TagFilterData[tagFilter].has(tag.key));
+  }
+}
+
 function Tags({event, projectSlug}: Props) {
   const [tagFilter, setTagFilter] = useState<TagFilter>(TagFilter.ALL);
-  const handleTagFilterChange = useCallback((value: TagFilter) => {
-    setTagFilter(value);
-  }, []);
+  const [tags, setTags] = useState<EventTag[]>(filterTags(event.tags ?? [], tagFilter));
+  const handleTagFilterChange = useCallback(
+    (value: TagFilter) => {
+      setTagFilter(value);
+      setTags(() => filterTags(event.tags, value));
+    },
+    [event]
+  );
 
   const hasNewTagsUI = useHasNewTagsUI();
+  const availableFilters = Object.keys(TagFilterData).filter(filter => {
+    return event.tags.some(tag => TagFilterData[filter].has(tag.key));
+  });
```

Comment:
nit - not doing anything with the keys
```suggestion
  const availableFilters = Object.values(TagFilterData).filter(filterSet => {
    return event.tags.some(tag => filterSet.has(tag.key));
  });
```